gh-101100: Fix Sphinx warning in references with asterisks (#113029)
python/cpython: Doc/library/socket.rst

.. method:: socket.getsockopt(level, optname[, buflen])

   Return the value of the given socket option (see the Unix man page

>    :manpage:`getsockopt(2)`).  The needed symbolic constants (:const:`!SO_\*` etc.)

Reviewer: ```suggestion
   :manpage:`getsockopt(2)`).  The needed symbolic constants (:ref:`SO_\* etc. <socket-unix-constants>`)
```

   Set the value of the given socket option (see the Unix manual page

>    :manpage:`setsockopt(2)`).  The needed symbolic constants are defined in the

Reviewer: ```suggestion
   :manpage:`setsockopt(2)`).  The needed symbolic constants are defined in this
```

>    :mod:`socket` module (:const:`!SO_\*` etc.).  The value can be an integer,

Reviewer: ```suggestion
   module (:ref:`!SO_\* etc. <socket-unix-constants>`).  The value can be an integer,
```
